library/Mockery/MockInterface.php

```diff
@@ -81,6 +81,21 @@ public function shouldDeferMissing();
     public function makePartial();
 
     /**
+     * Disable mocking non-existent methods
+     *
+     * @return Mock
+     */
+    public function disallowMockingNonExistentMethods();
```

Comment:
@GrahamCampbell I got your point.
The `disallowMockingNonExistentMethods()` can be removed from the interface and become a unique method for `Mockery\Mock` class.

But for `Mock` class to be able to locally (per mock basis, without setting it from global configuration) disallow mocking non existent methods, the `Mockery` class needs to have a way to read the local flag. Thus, adding `mockingNonExistentMethodsAllowed()` to the `MockInterface` is inevitable.

I'll try to look for a better way implementing this. Hopefully I can find one that doesn't need to change the interface :D